src/mail/view/MailViewerToolbar.ts

```diff
@@ -152,9 +155,13 @@ export class MailViewerActions implements Component<MailViewerToolbarAttrs> {
 								"{current}": Math.round((operation.progress() / 100) * attrs.mails.length).toFixed(0),
 								"{total}": attrs.mails.length,
 							}),
-						exportMails(attrs.mails, locator.mailFacade, locator.entityClient, locator.fileController, operation.id, ac.signal).finally(
-							operation.done,
-						),
+						exportMails(attrs.mails, locator.mailFacade, locator.entityClient, locator.fileController, operation.id, ac.signal)
+							.then((result) => {
+								if (result && result.length > 0) {
+									this.renderExportFailedEmailsDialog(result)
```

Comment:
convention-wise I would call this showExportFailedEmailsDialog since it's not part of the normal UI hierarchy in this component.

it also doesn't need to be a method as far as I can see